Automattic/wp-calypso
# Purchases: Replace the cancellation confirmation screen with a dialog

This pull request addresses part of #2392 by opening a dialog instead of showing the confirmation cancellation for all non-domain-registration subscriptions within their refund period.

When you cancel within the refund period you should see:
<img width="631" alt="screen shot 2016-04-01 at 17 54 37" src="https://cloud.githubusercontent.com/assets/275961/14214084/ffc7dadc-f832-11e5-8fe2-bde0019b3c31.png">

When you confirm this action, you should see:
<img width="862" alt="screen shot 2016-04-01 at 17 54 43" src="https://cloud.githubusercontent.com/assets/275961/14214086/0177c324-f833-11e5-86da-1ab7d1c10109.png">
 
#### Testing instructions - refund

3. Find a purchase within the refund period (not a domain)
4. Try to cancel the purchase
5. Assert that you don't see the confirmation cancellation screen
6. Assert that you are shown a Dialog before the cancellation is processed
6. Submit the Dialog
7. Assert that you see the Success notice telling you to check your email

#### Testing instructions - no refund

3. Find a purchase outside the refund period
4. Try to cancel the purchase
5. Assert that you see the Success notice telling you to check your email

#### Testing instructions - domain

3. Find a domain within the refund period
4. Try to cancel the purchase
5. Assert that you are still taken to the confirmation cancellation screen
6. Submit the confirmation cancellation screen
7. Assert that you see the Success notice telling you to check your email

#### Reviews
